### PR TITLE
Python: Add support for global attribute writes

### PR DESCRIPTION
**Pull request by tausbn**
I'm still not convinced that this is the best solution, but it fixes the
issue in the short term. When we eventually clean up the whole
`EssaNode` business, we can consider whether there's a better way to do
this.

--- 

I don't think this requires a change note, but will happily add one if needed.

### CODE REVIEW

**Comment by yoff** [comment]
Ideally, we would add a test showing the missed flow and then see the situation improve by the widened attribute write.

**Comment by tausbn** [comment]
> Ideally, we would add a test showing the missed flow and then see the situation improve by the widened attribute write.

Ah, good point. I'll add one.

File: python/ql/test/experimental/dataflow/typetracking/attribute_tests.py [hunk 1]
```diff
@@ -49,6 +49,22 @@ def test_create_with_foo():
...
+def test_global_attribute_read():
+    x = global_var.foo # $ tracked tracked=foo
+
+def test_local_attribute_assignment():
+    # Same as `test_global_attribute_assignment`, but the assigned variable is not global
+    local_var = object() # $ tracked=foo
+    local_var.foo = tracked # $ tracked tracked=foo
+
+def test_local_attribute_read():
+    x = local_var.foo
```
[the remark this answers is not in the document]
Comment (tausbn) [reply python/ql/test/experimental/dataflow/typetracking/attribute_tests.py:65]: The intent was to show that even though we have an attribute assignment to a (local) variable that shadows a global of the same name, this _does not_ result in additional flow. I can add `global local_var` if you think that'll make it clearer (though it is redundant). Assigning to the variable inside the function would make it a local variable, which would prevent the `ModuleVariableNode` from even existing.

[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/typetracking/attribute_tests.py:65]: Ah, the shadowing is an extra twist.

File: python/ql/test/experimental/dataflow/typetracking/attribute_tests.py [hunk 2]
```diff
@@ -49,6 +49,22 @@ def test_create_with_foo():
...
 
+def test_global_attribute_assignment():
+    global global_var
+    global_var.foo = tracked # $ tracked tracked=foo
+
+def test_global_attribute_read():
+    x = global_var.foo # $ tracked tracked=foo
+
+def test_local_attribute_assignment():
+    # Same as `test_global_attribute_assignment`, but the assigned variable is not global
```
[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/typetracking/attribute_tests.py:60]: Yes, I think so.

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**